tests/foundation.test.ts

```diff
@@ -0,0 +1,688 @@
+import * as fs from 'fs';
+import path from 'path';
+import { Contract, JWKInterface, PstState } from 'warp-contracts';
+
+import {
+  ActiveTier,
+  DelayedEvolveInput,
+  IOState,
+  ServiceTier,
+} from '../src/types';
+import { arweave, warp } from './setup.jest';
+import {
+  DEFAULT_FOUNDATION_ACTION_ACTIVE_STATUS,
+  DEFAULT_FOUNDATION_ACTION_FAILED_STATUS,
+  DEFAULT_FOUNDATION_ACTION_PASSED_STATUS,
+  DEFAULT_FOUNDATION_DELAYED_EVOLVE_COMPLETED_STATUS,
+  DEFAULT_INVALID_TIER_MESSAGE,
+  DEFAULT_MINIMUM_ALLOWED_EVOLUTION_DELAY,
+} from './utils/constants';
+import {
+  getCurrentBlock,
+  getLocalArNSContractId,
+  getLocalWallet,
+  mineBlock,
+  mineBlocks,
+} from './utils/helper';
+
+describe('FoundationAction', () => {
+  let contract: Contract<PstState>;
+  let foundationMember: JWKInterface;
+  let foundationMemberAddress: string;
+  let srcContractId: string;
+  let fees: { [x: string]: number };
+  let newLocalSourceCodeJS;
+  const newTier: ServiceTier = {
+    fee: 100,
+    settings: {
+      maxUndernames: 100,
+    },
+  };
+
+  beforeAll(async () => {
+    srcContractId = getLocalArNSContractId();
+    srcContractId = getLocalArNSContractId();
+    newLocalSourceCodeJS = fs.readFileSync(
+      path.join(__dirname, '../dist/contract.js'),
+      'utf8',
+    );
+  });
+
+  describe('valid foundation member', () => {
+    let newFoundationMember1: JWKInterface;
+    let newFoundationMemberAddress1: string;
+    let removedMember: JWKInterface;
+    let removedMemberAddress: string;
+    let newTierId: string;
+
+    beforeAll(async () => {
+      foundationMember = getLocalWallet(7);
+      foundationMemberAddress = await arweave.wallets.getAddress(
+        foundationMember,
+      );
+      newFoundationMember1 = getLocalWallet(8);
+      newFoundationMemberAddress1 = await arweave.wallets.getAddress(
+        newFoundationMember1,
+      );
+      removedMember = getLocalWallet(9);
+      removedMemberAddress = await arweave.wallets.getAddress(removedMember);
+      contract = warp.pst(srcContractId).connect(foundationMember);
+      fees = ((await contract.readState()).cachedValue.state as IOState).fees;
+    });
+
+    it('should initiate and complete add address', async () => {
+      const type = 'addAddress';
+      const id1 = 0;
+      const id2 = 1;
+      const target1 = newFoundationMemberAddress1;
+      const target2 = removedMemberAddress;
+      const note1 = 'Adding member 2';
+      const note2 = 'Adding member 3';
+      const writeInteraction1 = await contract.writeInteraction({
+        function: 'foundationAction',
+        type,
+        value: target1,
+        note: note1,
+      });
+      const start1 = await getCurrentBlock(arweave);
+      const writeInteraction2 = await contract.writeInteraction({
+        function: 'foundationAction',
+        type,
+        value: target2,
+        note: note2,
+      });
```

Comment:
for the sake of this test - we only need the one write interaction

Reply:
Yes, this adds an extra step in this specific test of addAddress.  But I do it so we have 2 members knowing that one will be removed and then using the other member to test 2-signature actions.  Can refactor this later with better multi-signature tests.